Makefile

```diff
@@ -19,26 +22,27 @@ check_isort:
 check_flake8:
 	flake8 $(python-packages)
 
-check_pylint: install_parser
+check_pylint: $(parser-installed)
 	pylint $(python-packages)
 
-check_mypy: install_parser
+check_mypy: $(parser-installed)
 	mypy --pretty $(python-packages)
 
-check_contracts:
-	pyicontract-lint $(python-packages)
-
 format:
 	black -l 100 $(python-packages)
 	isort $(python-packages)
 
-test: check test_python
+test: check test_python_coverage
+
+test_python: $(parser-installed)
+	python3 -m pytest -n$(shell nproc)  -vv --ignore=tests/data tests
 
-test_python: install_parser
-	python3 -m pytest -n$(shell nproc)  -vv -m "not hypothesis" tests
+test_python_coverage:
+	python3 -m pytest -n$(shell nproc) -vv --cov=librflxlang --cov-branch --cov-fail-under=71 --cov-report=term-missing:skip-covered --ignore=tests/data tests
 
-install_parser:
-	pip3 install .[Devel]
+$(parser-installed):
+	pip3 install --user .[Devel]
+	@touch $@
```

Comment:
1. The `--user` flag prevents the use of virtual environments which are commonly used for Python development for various reasons.
2. IMHO `pip install` shouldn't be automatically executed. It is rather unexpected that packages get installed when running tests. I would prefer moving the parser installation into a separate target and describing the necessary steps in a developer documentation.

Reply:
I made this a manual step and documented it in the README in d64ef8354f7328cd2ee40b3a77fded360f5c8a98.